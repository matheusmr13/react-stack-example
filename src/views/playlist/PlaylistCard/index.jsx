import React from 'react';

import './PlaylistCard.css';

const PlaylistCard = ({ playlist }) => {
	return (
		<div className="playlist-card">
			<img src={playlist.images[0].url} alt={playlist.name} />
			<div className="playlist-card__name">{playlist.name}</div>
		</div>
	);
};

export default PlaylistCard;
