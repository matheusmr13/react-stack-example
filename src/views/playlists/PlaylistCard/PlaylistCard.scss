.playlist-card {
	position: relative;
	display: inline-block;
	cursor: pointer;

	&__name {

	}

	&:hover {
		&::after {
			content: '';
			position: absolute;
			top: 0;
			left: 0;
			width: 100%;
			height: 100%;
			background-color: rgba(0, 0, 0, 0.3);
		}
	}
}