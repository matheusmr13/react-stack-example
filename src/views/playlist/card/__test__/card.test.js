import React from 'react';
import { shallow } from 'enzyme';
import PlaylistCard from 'views/playlist/card';

const playlist = {
	external_urls: {
		spotify: 'mylink'
	},
	id: '6ftJBzU2LLQcaKefMi7ee7',
	images: [{
		url: 'myotherlink'
	}],
	name: 'my playlist',
	owner: {
		external_urls: {
			spotify: 'owner link'
		},
		id: 'ownerid'
	}
};

describe('PlaylistCard component', () => {
	it('should render properly', () => {
		const playlistCard = shallow(<PlaylistCard playlist={playlist} />);
		expect(playlistCard).toMatchSnapshot();
	});
});
