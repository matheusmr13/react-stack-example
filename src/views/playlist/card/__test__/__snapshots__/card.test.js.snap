// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaylistCard component should render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PlaylistCard
    playlist={
        Object {
            "external_urls": Object {
              "spotify": "mylink",
            },
            "id": "6ftJBzU2LLQcaKefMi7ee7",
            "images": Array [
              Object {
                "url": "myotherlink",
              },
            ],
            "name": "my playlist",
            "owner": Object {
              "external_urls": Object {
                "spotify": "owner link",
              },
              "id": "ownerid",
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <CardShimmer />,
        <div
          style={
                    Object {
                              "display": "none",
                            }
          }
>
          <div
                    className="playlist-card__image"
          >
                    <img
                              alt="my playlist"
                              onLoad={[Function]}
                              src="myotherlink"
                    />
                    <div
                              className="playlist-card__hover"
                    >
                              <div
                                        className="circle"
                              >
                                        <svg
                                                  className="icon-play"
                                                  viewBox="0 0 85 100"
                                        >
                                                  <path
                                                            d="M81 44.6c5 3 5 7.8 0 10.8L9 98.7c-5 3-9 .7-9-5V6.3c0-5.7 4-8 9-5l72 43.3z"
                                                            fill="#fff"
                                                  >
                                                            <title>
                                                                      PLAY
                                                            </title>
                                                  </path>
                                        </svg>
                              </div>
                    </div>
          </div>
          <div
                    className="playlist-card__name"
          >
                    my playlist
          </div>
</div>,
      ],
      "className": "playlist-card",
      "href": "mylink",
      "target": "_blank",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="playlist-card__image"
>
              <img
                            alt="my playlist"
                            onLoad={[Function]}
                            src="myotherlink"
              />
              <div
                            className="playlist-card__hover"
              >
                            <div
                                          className="circle"
                            >
                                          <svg
                                                        className="icon-play"
                                                        viewBox="0 0 85 100"
                                          >
                                                        <path
                                                                      d="M81 44.6c5 3 5 7.8 0 10.8L9 98.7c-5 3-9 .7-9-5V6.3c0-5.7 4-8 9-5l72 43.3z"
                                                                      fill="#fff"
                                                        >
                                                                      <title>
                                                                                    PLAY
                                                                      </title>
                                                        </path>
                                          </svg>
                            </div>
              </div>
</div>,
            <div
              className="playlist-card__name"
>
              my playlist
</div>,
          ],
          "style": Object {
            "display": "none",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt="my playlist"
                  onLoad={[Function]}
                  src="myotherlink"
/>,
                <div
                  className="playlist-card__hover"
>
                  <div
                                    className="circle"
                  >
                                    <svg
                                                      className="icon-play"
                                                      viewBox="0 0 85 100"
                                    >
                                                      <path
                                                                        d="M81 44.6c5 3 5 7.8 0 10.8L9 98.7c-5 3-9 .7-9-5V6.3c0-5.7 4-8 9-5l72 43.3z"
                                                                        fill="#fff"
                                                      >
                                                                        <title>
                                                                                          PLAY
                                                                        </title>
                                                      </path>
                                    </svg>
                  </div>
</div>,
              ],
              "className": "playlist-card__image",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "my playlist",
                  "onLoad": [Function],
                  "src": "myotherlink",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="circle"
>
                    <svg
                                        className="icon-play"
                                        viewBox="0 0 85 100"
                    >
                                        <path
                                                            d="M81 44.6c5 3 5 7.8 0 10.8L9 98.7c-5 3-9 .7-9-5V6.3c0-5.7 4-8 9-5l72 43.3z"
                                                            fill="#fff"
                                        >
                                                            <title>
                                                                                PLAY
                                                            </title>
                                        </path>
                    </svg>
</div>,
                  "className": "playlist-card__hover",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <svg
                      className="icon-play"
                      viewBox="0 0 85 100"
>
                      <path
                                            d="M81 44.6c5 3 5 7.8 0 10.8L9 98.7c-5 3-9 .7-9-5V6.3c0-5.7 4-8 9-5l72 43.3z"
                                            fill="#fff"
                      >
                                            <title>
                                                                  PLAY
                                            </title>
                      </path>
</svg>,
                    "className": "circle",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <path
                        d="M81 44.6c5 3 5 7.8 0 10.8L9 98.7c-5 3-9 .7-9-5V6.3c0-5.7 4-8 9-5l72 43.3z"
                        fill="#fff"
>
                        <title>
                                                PLAY
                        </title>
</path>,
                      "className": "icon-play",
                      "viewBox": "0 0 85 100",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <title>
                          PLAY
</title>,
                        "d": "M81 44.6c5 3 5 7.8 0 10.8L9 98.7c-5 3-9 .7-9-5V6.3c0-5.7 4-8 9-5l72 43.3z",
                        "fill": "#fff",
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "PLAY",
                        },
                        "ref": null,
                        "rendered": "PLAY",
                        "type": "title",
                      },
                      "type": "path",
                    },
                    "type": "svg",
                  },
                  "type": "div",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "my playlist",
              "className": "playlist-card__name",
            },
            "ref": null,
            "rendered": "my playlist",
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "a",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <CardShimmer />,
          <div
            style={
                        Object {
                                    "display": "none",
                                  }
            }
>
            <div
                        className="playlist-card__image"
            >
                        <img
                                    alt="my playlist"
                                    onLoad={[Function]}
                                    src="myotherlink"
                        />
                        <div
                                    className="playlist-card__hover"
                        >
                                    <div
                                                className="circle"
                                    >
                                                <svg
                                                            className="icon-play"
                                                            viewBox="0 0 85 100"
                                                >
                                                            <path
                                                                        d="M81 44.6c5 3 5 7.8 0 10.8L9 98.7c-5 3-9 .7-9-5V6.3c0-5.7 4-8 9-5l72 43.3z"
                                                                        fill="#fff"
                                                            >
                                                                        <title>
                                                                                    PLAY
                                                                        </title>
                                                            </path>
                                                </svg>
                                    </div>
                        </div>
            </div>
            <div
                        className="playlist-card__name"
            >
                        my playlist
            </div>
</div>,
        ],
        "className": "playlist-card",
        "href": "mylink",
        "target": "_blank",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="playlist-card__image"
>
                <img
                                alt="my playlist"
                                onLoad={[Function]}
                                src="myotherlink"
                />
                <div
                                className="playlist-card__hover"
                >
                                <div
                                                className="circle"
                                >
                                                <svg
                                                                className="icon-play"
                                                                viewBox="0 0 85 100"
                                                >
                                                                <path
                                                                                d="M81 44.6c5 3 5 7.8 0 10.8L9 98.7c-5 3-9 .7-9-5V6.3c0-5.7 4-8 9-5l72 43.3z"
                                                                                fill="#fff"
                                                                >
                                                                                <title>
                                                                                                PLAY
                                                                                </title>
                                                                </path>
                                                </svg>
                                </div>
                </div>
</div>,
              <div
                className="playlist-card__name"
>
                my playlist
</div>,
            ],
            "style": Object {
              "display": "none",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <img
                    alt="my playlist"
                    onLoad={[Function]}
                    src="myotherlink"
/>,
                  <div
                    className="playlist-card__hover"
>
                    <div
                                        className="circle"
                    >
                                        <svg
                                                            className="icon-play"
                                                            viewBox="0 0 85 100"
                                        >
                                                            <path
                                                                                d="M81 44.6c5 3 5 7.8 0 10.8L9 98.7c-5 3-9 .7-9-5V6.3c0-5.7 4-8 9-5l72 43.3z"
                                                                                fill="#fff"
                                                            >
                                                                                <title>
                                                                                                    PLAY
                                                                                </title>
                                                            </path>
                                        </svg>
                    </div>
</div>,
                ],
                "className": "playlist-card__image",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "my playlist",
                    "onLoad": [Function],
                    "src": "myotherlink",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <div
                      className="circle"
>
                      <svg
                                            className="icon-play"
                                            viewBox="0 0 85 100"
                      >
                                            <path
                                                                  d="M81 44.6c5 3 5 7.8 0 10.8L9 98.7c-5 3-9 .7-9-5V6.3c0-5.7 4-8 9-5l72 43.3z"
                                                                  fill="#fff"
                                            >
                                                                  <title>
                                                                                        PLAY
                                                                  </title>
                                            </path>
                      </svg>
</div>,
                    "className": "playlist-card__hover",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <svg
                        className="icon-play"
                        viewBox="0 0 85 100"
>
                        <path
                                                d="M81 44.6c5 3 5 7.8 0 10.8L9 98.7c-5 3-9 .7-9-5V6.3c0-5.7 4-8 9-5l72 43.3z"
                                                fill="#fff"
                        >
                                                <title>
                                                                        PLAY
                                                </title>
                        </path>
</svg>,
                      "className": "circle",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <path
                          d="M81 44.6c5 3 5 7.8 0 10.8L9 98.7c-5 3-9 .7-9-5V6.3c0-5.7 4-8 9-5l72 43.3z"
                          fill="#fff"
>
                          <title>
                                                    PLAY
                          </title>
</path>,
                        "className": "icon-play",
                        "viewBox": "0 0 85 100",
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": <title>
                            PLAY
</title>,
                          "d": "M81 44.6c5 3 5 7.8 0 10.8L9 98.7c-5 3-9 .7-9-5V6.3c0-5.7 4-8 9-5l72 43.3z",
                          "fill": "#fff",
                        },
                        "ref": null,
                        "rendered": Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": "PLAY",
                          },
                          "ref": null,
                          "rendered": "PLAY",
                          "type": "title",
                        },
                        "type": "path",
                      },
                      "type": "svg",
                    },
                    "type": "div",
                  },
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "my playlist",
                "className": "playlist-card__name",
              },
              "ref": null,
              "rendered": "my playlist",
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "a",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
